### PR TITLE
fix: add `--preserve-env=PATH` to `sudo` command

### PR DESCRIPTION
**Pull request by dicej**
Previously, run_servers.sh would check that all the necessary prerequisites are
in the current shell's PATH, but since most of them are actually run via sudo,
what we really care about is whether those prerequisites are in *root's* PATH.
Now we check root's PATH for those things before proceeding.

This is important because the user may have installed Spin or Bindle using
`cargo install`, which only installs things for the current user, not system-wide.

Signed-off-by: Joel Dice <joel.dice@gmail.com>

### CODE REVIEW

File: run_servers.sh
Comment (bacongobbler) [inline run_servers.sh:28]: slight nitpick here, but can we put this above line 27 so the `require` lines are all sorted alphabetically?

File: run_servers.sh
Comment (bacongobbler) [inline run_servers.sh:23]: same here on alphabetical order - can we move this below `spin`?

**Review by bacongobbler — APPROVED** [review]
LGTM! Just had a few comments on alphabetical sorting. Not necessary to address before merging.

**Comment by dicej** [comment]
Good call -- they're now alphabetized.

**Review by bacongobbler — APPROVED** [review]
double LGTM - I'll let @adamreese merge this one as he's the primary maintainer of this project.

**Comment by bacongobbler** [comment]
No. This is required for production. Nomad must run as root in Linux for the `exec` driver to work.

https://www.nomadproject.io/docs/drivers/exec#client-requirements

Because `nomad` is executed as root, any jobs it spawns runs as root, so the binaries for `bindle`, `traefik`, and `spin` all require to be in the root user's PATH.

**Comment by bacongobbler** [comment]
For MacOS users, this is irrelevant because `raw_exec` does not require to be run as root.

https://www.nomadproject.io/docs/drivers/raw_exec#client-requirements

**Comment by adamreese** [comment]
In what cases would these not be in sudo's path?

**Comment by bacongobbler** [comment]
> This is important because the user may have installed Spin or Bindle using
> `cargo install`, which only installs things for the current user, not system-wide.

**Comment by bacongobbler** [comment]
I commonly place dev tools in `$HOME/bin`. In those cases, these binaries would not be accessible by root's `$PATH`.

**Comment by dicej** [comment]
Just force-pushed a different approach:

> fix: add `--preserve-env=PATH` to `sudo` command
    
> This ensures that Nomad inherits the current user's `PATH` environment variable,
> rather that the default `sudo` behavior of using a sanitized PATH.
    
> This is necessary because the user may have installed `bindle-server`,
> `traefik`, etc. locally but not system-wide.  For example, `cargo install`
> installs in the current user's home directory by default.

**Pull request merged**
